components/match2/wikis/warcraft/match_group_input_custom.lua

```diff
@@ -78,682 +138,418 @@ function CustomMatchGroupInput._readDate(matchArgs)
-					Variables.varDefault(name .. '_flag')
-				)),
-			extradata = {faction = Faction.read(Logic.emptyOr(
-					opponent['p' .. playerIndex .. 'race'],
-					Variables.varDefault(name .. '_faction')
-				)) or Faction.defaultFaction}
-		})
-	end
-
-	table.sort(links)
-
-	return {
-		type = opponent.type,
-		name = table.concat(links, ' / '),
-		score = opponent.score,
-		extradata = opponent.extradata,
-		match2players = players
-	}
-end
+	map.finished = MapFunctions.isFinished(mapInput, opponentCount)
+	local opponentInfo = Array.map(Array.range(1, opponentCount), function(opponentIndex)
+		local score, status = MatchGroupInputUtil.computeOpponentScore({
+			walkover = mapInput.walkover,
+			winner = mapInput.winner,
+			opponentIndex = opponentIndex,
+			score = mapInput['score' .. opponentIndex],
+		}, MapFunctions.calculateMapScore(mapInput.winner, map.finished))
+		return {score = score, status = status}
+	end)
 
----@param opponent table
----@param date string
----@return table
-function CustomMatchGroupInput.ProcessTeamOpponentInput(opponent, date)
-	local template = string.lower(Logic.emptyOr(opponent.template, opponent[1], '')--[[@as string]]):gsub('_', ' ')
+	map.scores = Array.map(opponentInfo, Operator.property('score'))
 
-	if String.isEmpty(template) or template == 'noteam' then
-		opponent = Table.merge(opponent, Opponent.blank(Opponent.team))
-		opponent.name = Opponent.toName(opponent)
-		return opponent
+	if map.finished or MatchGroupInputUtil.isNotPlayed(map.winner, mapInput.finished) then
+		map.resulttype = MatchGroupInputUtil.getResultType(mapInput.winner, mapInput.finished, opponentInfo)
+		map.walkover = MatchGroupInputUtil.getWalkover(map.resulttype, opponentInfo)
+		map.winner = MatchGroupInputUtil.getWinner(map.resulttype, mapInput.winner, opponentInfo)
 	end
 
-	assert(mw.ext.TeamTemplate.teamexists(template), 'Missing team template "' .. template .. '"')
-
-	local templateData = mw.ext.TeamTemplate.raw(template, date)
-
-	opponent.icon = templateData.image
-	opponent.icondark = Logic.emptyOr(templateData.imagedark, templateData.image)
-	opponent.name = templateData.page:gsub(' ', '_')
-	opponent.template = templateData.templatename or template
-
-	return opponent
+	return map, subGroup
 end
 
---MapInput functions
-
----@param match table
----@param mapIndex integer
----@param subGroupIndex integer
----@return integer
-function CustomMatchGroupInput._mapInput(match, mapIndex, subGroupIndex)
-	local map = Json.parseIfString(match['map' .. mapIndex])
-	map.map = (MapsData[(map.map or ''):lower()] or {}).name or map.map
-
-	-- set initial extradata for maps
-	map.extradata = {
-		comment = map.comment or '',
-		header = map.header or '',
-	}
-
-	-- determine score, resulttype, walkover and winner
-	map = CustomMatchGroupInput._mapWinnerProcessing(map)
+---@param mapInput table
+---@param opponentCount integer
+---@return boolean
+function MapFunctions.isFinished(mapInput, opponentCount)
```

Comment:
Use commons for the stuff that is on commons?

Reply:
First part is the same, the bestof calc only triggers if map bestof is added

Reply:
can't use
```
	if MatchGroupInputUtil.mapIsFinished(map, opponents) then
		return true
	end

	-- do the score check here
```

because that would make `finished=false` input invalid

Reply:
Sure, would save some dups at least